src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs

```diff
@@ -697,6 +791,8 @@ private int BodyRead(byte[] buffer, int offset, int count)
 						}
 					}
 					break;
+				default:
+					throw new InvalidOperationException("Internal Error: Unsupported compression method encountered.");
```

Comment:
do we ever allow it to get to this point for an entry with an unsupported compression method? (maybe it makes sense to check in any case)

Reply:
No it should never get to this point.

This is just precaution in case there would ever an additional compression method so that it's not forgotten to be extended here.